company.el

```diff
@@ -1340,7 +1340,14 @@ can retrieve meta-data for them."
             (and (symbolp this-command)
                  (string-match-p "\\`company-" (symbol-name this-command)))))))
 
+(defvar company-auto-update-doc nil
+  "If non-nil, company updates the doc-buffer whenever the selection changes.
+Use `company-show-doc-buffer' to toggle this variable.")
+
 (defun company-call-frontends (command)
+  (when (and company-auto-update-doc
+             (memq command '(update show)))
+    (company-show-doc-buffer))
```

Comment:
I'm not quite happy about this call (it does something other than what the function name advertises), but the effect is probably what we want for this feature.

Looking good otherwise, thanks!

Reply:
One possibility to solve your concern is to refactor these three lines into a company-doc-frontend and append that to company-frontends or a new company-internal-frontends list. This solution, however, seems unjustifiably complicated to me.

Reply:
Two other, lighter options:

* Create a wrapper function for `company-call-frontends` which would start by calling this code.
* Create a helper with this code, which would be invoked explicitly before every call to `(company-call-frontends 'update)`. All 4 of them.

Reply:
The first one is more maintainable, I think.  But I'm unable to come up with a good name for the wrapper function.  `company-frontend-event` or `company-handle-event`?

Reply:
`company-frontends-update`? Maybe.

I couldn't think of a good name, that's why I didn't suggest this in the first place. :-)